apps/dashboard/src/components/activity/components/status-badge.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { Badge, BadgeVariant } from '@/components/primitives/badge';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/primitives/popover';
+import { IActivityJob, JobStatusEnum } from '@novu/shared';
+import { StatusPreviewCard } from './status-preview-card';
+import { JOB_STATUS_CONFIG } from '../constants';
+
+export interface StatusBadgeProps {
+  jobs: IActivityJob[];
+}
+
+export function StatusBadge({ jobs }: StatusBadgeProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const errorCount = jobs.filter((job) => job.status === JobStatusEnum.FAILED).length;
+  let hoverTimeout: NodeJS.Timeout;
+
+  const handleMouseEnter = () => {
+    clearTimeout(hoverTimeout);
+
+    hoverTimeout = setTimeout(() => {
+      setIsOpen(true);
+    }, 300);
+  };
+
+  const handleMouseLeave = () => {
+    clearTimeout(hoverTimeout);
+    setIsOpen(false);
+  };
+
+  const status = getActivityStatus(jobs);
+
+  const { variant, icon: Icon, label } = JOB_STATUS_CONFIG[status] || JOB_STATUS_CONFIG[JobStatusEnum.PENDING];
+  const displayLabel =
+    status === JobStatusEnum.FAILED ? `${errorCount} ${errorCount === 1 ? 'ERROR' : 'ERRORS'}` : label;
+
+  return (
+    <Popover open={isOpen}>
```

Comment:
When I hover over the popover content, it blinks.
https://github.com/user-attachments/assets/3c4f57be-4cbf-4922-91d6-d7b273afd457

Reply:
It might be related to the mouse enter/leave events. There is no option to delay the initial mouse enter, but maybe we can use the `Tooltip` here instead which has that option. But I'm not sure.

Reply:
Unfortuently tooltip is very different style-wise, i've implemented a simple timeout delay to handle the mouse transitioning and the flicker is now preveneted